### ISSUE TITLE
[HOLD for payment 2023-04-20] [$4000] Update RN version to our latest v0.71.2-alpha.3

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#15124]
### Problem

To fix some bugs without patches we need to update our RN fork to newer version, we had merged those changes to our fork https://github.com/Expensify/react-native/releases/tag/v0.71.2-alpha.1 and we would like to include that in the App

### Why

Includes some new fixes in the react-native fork

### Solution

Update App's RN version to our latest https://github.com/Expensify/react-native/releases/tag/v0.71.2-alpha.1

cc @roryabraham 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0151c999f86eec33a8</li>
        <li>Upwork Job ID: 1625450113750552576</li>
        <li>Last Price Increase: 2023-04-13</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @maddylewis (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0151c999f86eec33a8

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Triggered auto assignment to @laurenreidexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @Santhosh-Sellavel (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @chiragsalian (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by mountiny** [comment]
Unassigning Lauren as that was an autoassigner blip.

here is a draft PR inspiration https://github.com/Expensify/App/pull/14618

**Comment by priyeshshah11** [comment]
@mountiny How is the proposal process going to work for this? I would like to do it if it's going to be on a volunteer basis, because there is not much to propose here

**Comment by abdulrahuman5196** [comment]
@mountiny  Is this open for external contribution? If so I would volunteer to contribute?

**Comment by mountiny** [comment]
This is going to be involved and take lots of testing and ideally someone already has experience with updating the RN version. I was thinking of offering this to C+ because of the importance and non-triviality of this change.

@priyeshshah11 @abdulrahuman5196 Would you both want to collaborate on this? One can make the PR and the other one will be testing extensively and suggesting any improvements if need be?

In terms of the reward, if all goes well and we can merge this soon we got $1500 split between you two? Would that work?

for better collaboration you could discuss anything in Slack thread to speed up the progress. thoughts?

**Comment by mountiny** [comment]
Here is a WIP branch which can be inspiration https://github.com/Expensify/App/pull/14618

And here is a helper tool which you can use to know what to update also https://react-native-community.github.io/upgrade-helper/

**Comment by priyeshshah11** [comment]
I would be happy to collaborate & create the PR.

**Comment by abdulrahuman5196** [comment]
@mountiny  @priyeshshah11 Yes. I would be happy to collaborate and work. We can sync up over slack thread to discuss on tasks to move forward?

**Comment by MelvinBot** [comment]
📣 @priyeshshah11 You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by priyeshshah11** [comment]
@mountiny We will also need to update [react-native-onyx](https://github.com/Expensify/react-native-onyx/blob/main/package.json#L84) to update `react-native-quick-sqlite` peerDependency to `8.0.0-beta.2` as that has the react native `v0.71` upgrade.

**Comment by mountiny** [comment]
Can you make a PR for that? We should probably test that separately too. cc @roryabraham

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $2000

**Comment by mountiny** [comment]
Increased the price due to the complexity of this issue

**Comment by MelvinBot** [comment]
@mountiny, @maddylewis, @priyeshshah11, @abdulrahuman5196, @Santhosh-Sellavel Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by mountiny** [comment]
This is still being worked on, resolving issues as we stumble upon them

**Comment by mountiny** [comment]
Janic has jumped in to the PR with helpful comments, we will have to update our fork again most likely

**Comment by MelvinBot** [comment]
@mountiny, @maddylewis, @priyeshshah11, @abdulrahuman5196, @Santhosh-Sellavel Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by priyeshshah11** [comment]
Hi @maddylewis, we are waiting on [this PR](https://github.com/Expensify/react-native/pull/46) to be merged to release a new version of our RN fork & then do further testing.

**Context:** https://github.com/Expensify/App/pull/15130#issuecomment-1472881247 & https://github.com/Expensify/react-native/pull/46#pullrequestreview-1346299037

CC: @mountiny @roryabraham @abdulrahuman5196 @janicduplessis

**Comment by mountiny** [comment]
Thanks Priyesh!

**Comment by roryabraham** [comment]
https://github.com/Expensify/react-native/pull/46 has been merged and a new version has been released, so we should be unblocked here.

**Comment by mountiny** [comment]
Still working on the PR

**Comment by tienifr** [comment]
Hi, @priyeshshah11 @Santhosh-Sellavel Please help to confirm that we need a PR to upgrade the version of React-native-ONYX in the App repo for this issue. It is blocking [another PR](https://github.com/Expensify/App/pull/16531)

**Comment by priyeshshah11** [comment]
@tienifr I didn't understand your question? We are upgrading react-native-onyx version from 1.0.36 to 1.0.38 in this PR, do you want us to upgrade it to v1.0.39?

**Comment by tienifr** [comment]
@priyeshshah11 Now, the version of react-native-onyx in App repo iss 1.0.36. Could you confirm that this issue will upgrade react-native-onyx version from 1.0.36 to 1.0.38 in App repo?

**Comment by priyeshshah11** [comment]
@tienifr yes this PR is on the App repo so yes it will upgrade react-native-onyx version from 1.0.36 to 1.0.38

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $3000

**Comment by mountiny** [comment]
Increasing the price here as its a major lift.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.99-6 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15130


If no regressions arise, payment will be issued on 2023-04-20. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Santhosh-Sellavel / @mountiny] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Santhosh-Sellavel / @mountiny] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Santhosh-Sellavel / @mountiny] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@maddylewis] Determine if we should create a regression test for this bug.
- [ ] [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@maddylewis] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by mountiny** [comment]
No regression tests required, this was not regression from nothing as well.

@abdulrahuman5196 @priyeshshah11 Are you happy to split the reward as $1000 to Abdul and $2000 to Priyesh? Its been a lot of work but I feel like Priyesh had a bit more work to do here, feel free to propose otherwise.

There will be also $1000 to @aimane-chnaif and $1000 @Santhosh-Sellavel as C+ review and testing, does this work?

**Comment by abdulrahuman5196** [comment]
@mountiny I too agree. @priyeshshah11 did good amount of work. I am fine with the split up.

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $4000

**Comment by Pujan92** [comment]
@priyeshshah11 If I am pointing correctly 2 issues of icons are raised after this PR change where in android Text -> View -> svg are not rendering properly.

https://github.com/Expensify/App/issues/17368
https://github.com/Expensify/App/issues/17433

**Comment by priyeshshah11** [comment]
> @priyeshshah11 If I am pointing correctly 2 issues of icons are raised after this PR change where in android Text -> View -> svg are not rendering properly.
> 
> #17368 #17433

@Pujan92 I have been looking into these issues for past couple of days & I don't know what's exactly wrong nor I have a solution yet but I don't think it's caused by this PR. I also tried upgrading `react-native-svg` to v13.9.0 & `svgr/webpack` to v7.0.0 but no luck. It's weird as I can see the icons appear as soon as I save the files but they still appear in the wrong place. @mountiny maybe someone from the software mansion team can help here?

**Comment by Pujan92** [comment]
@priyeshshah11 Can this connect to the version of react-native? bcoz before posting here I switched to the earlier commit of the old version and reinstalled it with that and it seemed to work.

**Comment by Pujan92** [comment]
I just tried upgrading the version of react-native to 0.71.6 and it showing the icons, maybe you can just give it a try once to confirm

<img width="650" alt="Screenshot 2023-04-18 at 6 10 16 PM" src="https://user-images.githubusercontent.com/14358475/232782462-dc33fd4a-a621-4f91-9048-5753c21ef49a.png">

**Comment by mountiny** [comment]
Feel free to bring those issues to Slack @Pujan92 we can tag people who can help with it then, that will be easier

**Comment by maddylewis** [comment]
Let me know if the following is accurate before I process payment - thanks!

payments:
- @priyeshshah11: $2750
- @abdulrahuman5196: $1250
- @Santhosh-Sellavel: $1000
- @aimane-chnaif: $1000

**Comment by mountiny** [comment]
looking good, thanks everyone!

**Comment by priyeshshah11** [comment]
@maddylewis I got the offer but I can't accept it as it shows as withdrawn, could you please resend?

**Comment by abdulrahuman5196** [comment]
@maddylewis  same here

**Comment by maddylewis** [comment]
i hadn't paid it yet. that just happened so you should be all set.

**Comment by priyeshshah11** [comment]
> i hadn't paid it yet. that just happened so you should be all set.

still the same here, can't accept the offer

**Comment by priyeshshah11** [comment]
accepted, thanks! @maddylewis

**Comment by abdulrahuman5196** [comment]
@maddylewis Seems this issue is closed. I have accepted the upwork request, but haven't got paid out.

**Comment by maddylewis** [comment]
@abdulrahuman5196 - just paid!
